lib/shopify_app/controller_concerns/login_protection.rb

```diff
@@ -85,5 +85,11 @@ def sanitize_shop_param(params)
       return unless params[:shop].present?
       ShopifyApp::Utils.sanitize_shop_domain(params[:shop])
     end
+
+    def sanitized_params
+      query_params = request.query_parameters
+      query_params[:shop] = sanitize_shop_param(params) if params[:shop].present?
```

Comment:
I think this should only happen if `params[:shop]` is a string similar to our [check here](https://github.com/Shopify/shopify_app/blob/3fb589d71bc03a11a8bb48bf87e613f6ce0210ea/lib/shopify_app/controller_concerns/login_protection.rb#L30).

When `params[:shop]` is not a string, that means that it could be a hash. This will happen when someone has a `Shop` model in their Rails app, and there are some parameters being passed for it.